src/Error/SerializationError.php

```diff
@@ -0,0 +1,17 @@
+<?php
+
+declare(strict_types=1);
+
+namespace GraphQL\Error;
+
+use Exception;
+
+/**
+ * Thrown when failing to serialize a scalar value.
+ *
+ * Not generally safe for clients, as the wrong given value could
+ * be something not intended to ever be seen by clients.
+ */
+class SerializationError extends Exception
```

Comment:
Shouldn't we name it ScalarSerializationError?

or even `ScalarSerializationFailed extends Exception`?

Reply:
I chose this name because serialization is not limited to scalars, but rather general to leaf types. Your comment did remind me to use it in `EnumType` as well, thank you.